src/common/log.c

```diff
@@ -0,0 +1,267 @@
+#include "common/log.h"
+
+#include <string.h>
+
+#if defined(_WIN32) || defined(_WIN64)
+#include <windows.h>
+#else
+#include <unistd.h>
+#endif
+
+#include "common/log_file.h"
+
+// Configure log decoration, see zf_log.c
+#define ZF_LOG_MESSAGE_CTX_FORMAT (\
+	HOUR, S(":"), MINUTE, S(":"), SECOND, S("."), MILLISECOND, S(ZF_LOG_DEF_DELIMITER), \
+	LEVEL, S(ZF_LOG_DEF_DELIMITER) \
+)
+
+// Return minimum of comparable arguments a and b
+#define MIN(a,b) (a <= b ? a : b)
+
+// Retry POSIX syscall until it returns non-zero
+#define RETVAL_UNUSED(expr) \
+	do { while(expr) break; } while(0)
+
+/*
+ * Discard Log-writing callback
+ *
+ * Before we call ardop_log_start(), all log messages are discarded
+ * by default.
+ */
+static void log_callback_discard(const zf_log_message* msg, void* param);
+
+/*
+ * Log-writing callback
+ *
+ * Receives formatted messages from zf_log for printing. This method
+ * is based on zf_log_out_stderr_callback from zf_log.c.
+ */
+static void log_callback(const zf_log_message* msg, void* param);
+
+/*
+ * Global log verbosity level
+ */
+int _zf_log_global_output_lvl = ZF_LOG_INFO;
+
+/*
+ * Global log output facility
+ *
+ * The default callback discards all messages until we enable them
+ * via ardop_log_start()
+ */
+zf_log_output _zf_log_global_output = { ZF_LOG_PUT_STD, 0, log_callback_discard };
+
+/*
+ * Console log verbosity
+ */
+static int ArdopLogVerbosityConsole = ZF_LOG_INFO;
+
+/*
+ * File log verbosity
+ */
+static int ArdopLogVerbosityFile = ZF_LOG_INFO;
```

Comment:
Previously FileLogLevel had a default value of LOGDEBUG, and I think that was a good choice.  So, I think that ArdopLogVerbosityFile should be ZF_LOG_DEBUG rather than ZF_LOG_INFO.  If I understand these settings correctly, that means that _zf_log_global_output_lvl must also be set to ZF_LOG_DEBUG.